src/com/itacademy/eshop/shop/ShoppingCart.java

```diff
@@ -24,6 +24,19 @@ public String getTotalPrice() {
         /**
          * Cakculates total price of shopping cart
          */
-        throw new UnsupportedOperationException("getTotalPrice() is not implemented yet");
+        double sum = 0;
+
+        for (Product product: this.products) {
```

Comment:
Usually _this_ keyword is used in setters, when you need to pass the whole object or call other constructors of a class.
Since you have only one already defined _products_ ArrayList in this class, use of _this_ keyword is not required.
Again, won't break anything, just not really required.